lib/client-retry.js

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2021 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+"use strict";
+const { AssetComputeClient } = require("./client");
+const colors = require("colors/safe");
+const sleep = require("util").promisify(setTimeout);
+
+const MAX_RETRY_429 = 4;
+const MAX_RETRY_WAIT_TIME = 60000;
+const MIN_RETRY_WAIT_TIME = 30000;
+
+/**
+ * Small wrapper around AssetComputeClient adding retry logic on 429s based on `retry-after` header
+ */
+class AssetComputeClientRetry extends AssetComputeClient {
+    constructor(integration, options) {
+        super(integration, options);
+        this.max429RetryCount = (options && options.max429RetryCount) || MAX_RETRY_429;
+    }
+
+    retryWaitTime(e) {
+        // randomly choose between 30-60 seconds waittime
+        let waitTime = Math.floor(Math.random() * (MAX_RETRY_WAIT_TIME - MIN_RETRY_WAIT_TIME + 1)) + MIN_RETRY_WAIT_TIME;
+
+        // in seconds amount of time to wait before retrying the request
```

Comment:
seconds or milliseconds? I have a doubt now - seems large to be in seconds

Reply:
sorry about the unclearness here
- api gateway `retry-after` header comes back in seconds
- every other variable is in ms

Reply:
(I only know this from testing, still on the hunt for the official api gateway docs)

Reply:
I guess I should take into account that it could be a date (says the doc), I will add this

Reply:
Alright done: https://github.com/adobe/asset-compute-client/pull/53/commits/80f1650a6cbcb4002685ee0b065092c9283e5c37
I created a new custom error class: `TooManyRequestsError`
It handles converting the stringified `retry-after` header (either as a date or number) into a number in seconds to wait until retrying